dipy/reconst/mapmri.py

```diff
@@ -267,21 +266,19 @@ def __init__(self,
                 raise ValueError('CVXPY package needed to enforce constraints.')
             if cvxpy_solver is not None:
                 if cvxpy_solver not in cvxpy.installed_solvers():
-                    msg = "Input `cvxpy_solver` was set to %s." % cvxpy_solver
-                    msg += " One of %s" % ', '.join(cvxpy.installed_solvers())
-                    msg += " was expected."
-                    raise ValueError(msg)
+                    installed_solvers = ', '.join(cvxpy.installed_solvers())
+                    raise ValueError(f"Input `cvxpy_solver` was set to"
```

Comment:
I do not think the f-string is necessary at this specific line. but I agree it is necessary below. 

There are others cases like that in your change, but that fine, I will go ahead and merge it. 

Thanks !

Reply:
You're right, it's not really necessary. [Horizontally align quotes in multi-line f-string breaks](https://github.com/psf/black/issues/2915) is a good summary of how I ended up adding `f` to lines that do not really need it, without being overly happy with it.

Reply:
I see, thanks for the pointer!